core/src/main/java/org/fao/geonet/util/XslUtil.java

```diff
@@ -135,6 +139,37 @@ public static String getSettingValue(String key) {
 	public static boolean existsBean(String beanId) {
 		return ProfileManager.existsBean(beanId);
 	}
+
+
+    /**
+     * Return value of property 'propName' of bean 'beanId'
+     *
+     * @param beanId Name of bean to extract property from
+     * @param propName Name of property
+     * @return String value of property 'propName' of bean 'BeanId'
+     */
+    public static String getBeanPropertyValue(String beanId, String propName) throws Exception {
+
+        if(!XslUtil.existsBean(beanId))
+            throw new Exception("Cannot find bean with name : " + beanId);
+
+        ServiceContext serviceContext = ServiceContext.get();
+
+        if (serviceContext == null)
+            throw new Exception("Cannot get ServiceContext instance");
+
+        ServletContext servletContext = serviceContext.getServlet().getServletContext();
+        ConfigurableApplicationContext springContext = JeevesDelegatingFilterProxy.getApplicationContextFromServletContext(servletContext);
+
+        Object o = springContext.getBean(beanId);
+        Field f = o.getClass().getDeclaredField(propName);
+        boolean isAccessible = f.isAccessible();
+        f.setAccessible(true);
+        String res = f.get(o).toString();
+        f.setAccessible(isAccessible);
```

Comment:
I wonder if this is useful to setAccessible() back to the original value. I tried the following groovy snippet:

```groovy
import java.lang.reflect.Field;
class Blah {
 private String field;
}
def b = new Blah();
Field f = b.getClass().getDeclaredField("field");
println f.isAccessible();

f.setAccessible(true);

f= b.getClass().getDeclaredField("field");
println f.isAccessible();

```
Which outputs the following:

```
false
false
```

My guess is that each time you will call getDeclaredField(), you will get a new object with the original "access state" of your members in your class. Anyway, this just argues that the line 169 can be removed, it does not harm though.